src/share/classes/com/alibaba/tenant/TenantResourceType.java

```diff
@@ -22,12 +22,153 @@
 package com.alibaba.tenant;
 
 import com.alibaba.rcm.ResourceType;
+import com.alibaba.rcm.Constraint;
+import java.util.stream.Collectors;
+import java.util.Arrays;
+import static com.alibaba.tenant.NativeDispatcher.*;
 
 /*
  * Type of resource that can be throttled when MultiTenant feature enabled
  */
 class TenantResourceType extends ResourceType {
+
+    /**
+     * Throttle CPU resource with cgroup controller: cpu.shares.
+     * Expecting one constraint value which represents cpu.shares value.
+     */
+    static final ResourceType CPU_SHARES = new TenantResourceType("CPU_SHARE",true) {
+        @Override
+        protected void validate(long... values) throws IllegalArgumentException {
+            if (values == null || values.length != 1
+                    || values[0] <= 0) {
+                throw new IllegalArgumentException("Bad CPU_SHARE constraint:" + values[0]);
+            }
+        }
+
+        @Override
+        public Constraint newConstraint(long ... values) {
+            validate(values);
+            return new JGroupConstraint(this, values) {
+                @Override
+                void sync(JGroup jgroup) {
+                    if (IS_CPU_SHARES_ENABLED) {
+                        jgroup.setValue(CG_CPU_SHARES, Integer.toString((int) getValues()[0]));
+                    }
+                }
+            };
+        }
+    };
+
+    /**
+     * Throttle CPU resource with cgroup controller: cpuset.cpus.
+     * Expecting N constraint value which represent cores described in cpuset.cpus.
+     * e.g. {@code TenantResourceType.CPUSET_CPUS.newConstraint(LongStream.range(0, 16).toArray()) }
+     */
+    static final ResourceType CPUSET_CPUS = new TenantResourceType("CPUSET_CPUS",
+            true) {
```

Comment:
make the two parameters of constructor method 'TenantResourceType' in the same line